egs/librispeech/ASR/pruned_transducer_stateless_multi_datasets/model.py

```diff
@@ -0,0 +1,249 @@
+           the form:
+              lm_scale * lm_probs + am_scale * am_probs +
+              (1-lm_scale-am_scale) * combined_probs
+        """
+        assert x.ndim == 3, x.shape
+        assert x_lens.ndim == 1, x_lens.shape
+        assert y.num_axes == 2, y.num_axes
+
+        assert x.size(0) == x_lens.size(0) == y.dim0
+
+        encoder_out, x_lens = self.encoder(x, x_lens)
+        assert torch.all(x_lens > 0)
+
+        # Now for the decoder, i.e., the prediction network
+        row_splits = y.shape.row_splits(1)
+        y_lens = row_splits[1:] - row_splits[:-1]
+
+        blank_id = self.decoder.blank_id
+        sos_y = add_sos(y, sos_id=blank_id)
+
+        # sos_y_padded: [B, S + 1], start with SOS.
+        sos_y_padded = sos_y.pad(mode="constant", padding_value=blank_id)
+
+        if libri:
+            decoder = self.decoder
+            joiner = self.joiner
+            simple_decoder_linear = self.simple_decoder_linear
+            simple_encoder_linear = self.simple_encoder_linear
+            decoder_linear = self.decoder_linear
+            encoder_linear = self.encoder_linear
+        else:
+            decoder = self.decoder_giga
+            joiner = self.joiner_giga
+            simple_decoder_linear = self.simple_decoder_giga_linear
+            simple_encoder_linear = self.simple_encoder_giga_linear
+            decoder_linear = self.decoder_giga_linear
+            encoder_linear = self.encoder_giga_linear
+
+        # decoder_out: [B, S + 1, C]
+        decoder_out = decoder(sos_y_padded)
+
+        # Note: y does not start with SOS
+        # y_padded : [B, S]
+        y_padded = y.pad(mode="constant", padding_value=0)
+
+        y_padded = y_padded.to(torch.int64)
+        boundary = torch.zeros(
+            (x.size(0), 4), dtype=torch.int64, device=x.device
+        )
+        boundary[:, 2] = y_lens
+        boundary[:, 3] = x_lens
+
+        simple_decoder_out = simple_decoder_linear(decoder_out)
+        simple_encoder_out = simple_encoder_linear(encoder_out)
+
+        simple_loss, (px_grad, py_grad) = k2.rnnt_loss_smoothed(
+            lm=simple_decoder_out,
+            am=simple_encoder_out,
+            symbols=y_padded,
+            termination_symbol=blank_id,
+            lm_only_scale=lm_scale,
+            am_only_scale=am_scale,
+            boundary=boundary,
+            reduction="sum",
+            return_grad=True,
+        )
+
+        # ranges : [B, T, prune_range]
+        ranges = k2.get_rnnt_prune_ranges(
+            px_grad=px_grad,
+            py_grad=py_grad,
+            boundary=boundary,
+            s_range=prune_range,
+        )
+
+        # am_pruned : [B, T, prune_range, C]
+        # lm_pruned : [B, T, prune_range, C]
+        am_pruned, lm_pruned = k2.do_rnnt_pruning(
+            am=simple_encoder_out, lm=simple_decoder_out, ranges=ranges
+        )
+
+        am_pruned = encoder_linear(am_pruned)
```

Comment:
Two `nn.Linear()` layers are used to transform the pruned outputs to the dimension of joiner_dim.